Mathlib/Topology/PartitionOfUnity.lean

```diff
@@ -174,6 +174,52 @@ theorem le_one (i : ι) (x : X) : f i x ≤ 1 :=
   (single_le_finsum i (f.locallyFinite.point_finite x) fun j => f.nonneg j x).trans (f.sum_le_one x)
 #align partition_of_unity.le_one PartitionOfUnity.le_one
 
+section finsupport
+/-- The support of a partition of unity at a point `x₀`
+  (i.e., the set of `i` such that `f i` doesn't vanish at `x₀`), as a `Finset`. -/
+def finsupport {s : Set X} (ρ : PartitionOfUnity ι X s) (x₀ : X) : Finset ι :=
+  (ρ.locallyFinite.point_finite x₀).toFinset
+
+@[simp]
+theorem coe_finsupport {s : Set X} (ρ : PartitionOfUnity ι X s) (x₀ : X) :
+    (ρ.finsupport x₀ : Set ι) = support fun i ↦ ρ i x₀ := by
+  dsimp only [finsupport]
+  rw [Finite.coe_toFinset]
+  rfl
+
+@[simp]
+theorem mem_finsupport {s : Set X} (ρ : PartitionOfUnity ι X s) (x₀ : X) {i} :
+    i ∈ ρ.finsupport x₀ ↔ i ∈ support fun i ↦ ρ i x₀ := by
+  simp only [finsupport, mem_support, Finite.mem_toFinset, mem_setOf_eq]
+
+theorem sum_finsupport {s : Set X} (ρ : PartitionOfUnity ι X s) {x₀ : X}
+    (hx₀ : x₀ ∈ s := by trivial) :
+    ∑ i in ρ.finsupport x₀, ρ i x₀ = 1 := by
+  have := ρ.sum_eq_one hx₀
+  rwa [finsum_eq_sum_of_support_subset] at this
+  rw [ρ.coe_finsupport]
+
+theorem sum_finsupport' {s : Set X} (ρ : PartitionOfUnity ι X s) {x₀ : X}
+    (hx₀ : x₀ ∈ s := by trivial) {I : Finset ι} (hI : ρ.finsupport x₀ ⊆ I) :
+    ∑ i in I, ρ i x₀ = 1 := by
+  classical
+  rw [← Finset.sum_sdiff hI, ρ.sum_finsupport hx₀]
+  suffices ∑ i in I \ ρ.finsupport x₀, ρ i x₀ = 0 by rw [this, zero_add]
+  suffices : ∑ i in I \ ρ.finsupport x₀, (ρ i) x₀ = ∑ i in I \ ρ.finsupport x₀, 0
+  rw [this, Finset.sum_const_zero]
+  apply Finset.sum_congr rfl
+  rintro x hx
+  simp only [Finset.mem_sdiff, ρ.mem_finsupport, mem_support, Classical.not_not] at hx
+  exact hx.2
+
+theorem sum_finsupport_smul {s : Set X} (ρ : PartitionOfUnity ι X s) {x₀ : X}
+    {M : Type _} [AddCommGroup M] [Module ℝ M] (φ : ι → X → M) :
+    ∑ i in ρ.finsupport x₀, ρ i x₀ • φ i x₀ = ∑ᶠ i, ρ i x₀ • φ i x₀ := by
+  apply (finsum_eq_sum_of_support_subset _ _).symm
+  erw [ρ.coe_finsupport x₀, support_smul]
```

Comment:
I slightly prefer avoiding the `erw`
```suggestion
  have : (fun i ↦ (ρ i) x₀ • φ i x₀) = (fun i ↦ (ρ i) x₀) • (fun i ↦ φ i x₀) := rfl
  rw [ρ.coe_finsupport x₀, this, support_smul]
```
but only slightly because I was hoping we had a lemma for what I've called `this` above but it seems not.